controllers/operatorpolicy_controller.go

```diff
@@ -1198,22 +1216,96 @@ func (r *OperatorPolicyReconciler) considerResolutionFailed(
 	}
 
 	if !includesSubscription {
+		if policy.Status.SubscriptionInterventionExpired() {
+			policy.Status.SubscriptionInterventionTime = nil
+		}
+
 		return mergedSub, nil, false, nil
 	}
 
 	// Handle non-ConstraintsNotSatisfiable reasons separately
 	if !strings.EqualFold(subResFailed.Reason, "ConstraintsNotSatisfiable") {
-		changed := updateStatus(policy, subResFailedCond(subResFailed), nonCompObj(foundSub, subResFailed.Reason))
+		changed := updateStatus(policy, subResFailedCond(subResFailed), nonCompObj(mergedSub, subResFailed.Reason))
+
+		if policy.Status.SubscriptionInterventionExpired() {
+			policy.Status.SubscriptionInterventionTime = nil
+		}
 
 		return mergedSub, nil, changed, nil
 	}
 
 	// A "constraints not satisfiable" message has nondeterministic clauses, and can be noisy with a list of versions.
 	// Just report a generic condition, which will prevent the OperatorPolicy status from constantly updating
 	// when the details in the Subscription status change.
-	changed := updateStatus(policy, subConstraintsNotSatisfiableCond, nonCompObj(foundSub, "ConstraintsNotSatisfiable"))
+	changed := updateStatus(policy, subConstraintsNotSatisfiableCond,
+		nonCompObj(mergedSub, "ConstraintsNotSatisfiable"))
+
+	if policy.Status.SubscriptionInterventionExpired() || policy.Status.SubscriptionInterventionTime == nil {
```

Comment:
Should this be after determining it is a ConstraintsNotSatisfiable that can be forcefully resolved?

Reply:
Good point, I should do the `unreferencedCSVRegex` thing earlier, I think.